lisdf/components/model.py

```diff
@@ -0,0 +1,224 @@
+    def from_quat_xyzw(cls, pos, xyzw) -> "Pose":
+        return cls(pos, np.array([xyzw[3], xyzw[0], xyzw[1], xyzw[2]]))
+
+    @classmethod
+    def identity(cls) -> "Pose":
+        return cls(
+            pos=np.zeros(3, dtype="float32"),
+            quat_wxyz=np.array([1, 0, 0, 0], dtype="float32"),
+        )
+
+    @cached_property
+    def quat_xyzw(self) -> Vector4f:
+        return np.array(
+            [self.quat_wxyz[1], self.quat_wxyz[2], self.quat_wxyz[3], self.quat_wxyz[0]]
+        )
+
+    @cached_property
+    def rpy(self) -> Vector3f:
+        return euler_from_quaternion(self.quat_xyzw)  # type: ignore
+
+
+@dataclass
+class Inertia(StringConfigurable):
+    ixx: float
+    ixy: float
+    ixz: float
+    iyy: float
+    iyz: float
+    izz: float
+
+    @classmethod
+    def zeros(cls) -> "Inertia":
+        return cls(0, 0, 0, 0, 0, 0)
+
+    @classmethod
+    def from_diagnal(cls, ixx, iyy, izz) -> "Inertia":
+        return cls(ixx, 0, 0, iyy, 0, izz)
+
+    @property
+    def matrix(self) -> np.ndarray:
+        return np.array(
+            [
+                [self.ixx, self.ixy, self.ixz],
+                [self.ixy, self.iyy, self.iyz],
+                [self.ixz, self.iyz, self.izz],
+            ],
+            dtype=np.float32,
+        )
+
+
+@dataclass
+class Inertial(StringConfigurable):
+    mass: float
+    pose: Pose
+    inertia: Inertia
+
+    @classmethod
+    def zeros(cls) -> "Inertial":
+        return cls(0, Pose.identity(), Inertia.zeros())
+
+
+@dataclass
+class SurfaceContact(StringConfigurable):
+    mjcf_configs: Optional[Dict[str, Any]] = None
+    sdf_configs: Optional[Dict[str, Any]] = None
+
+
+@dataclass
+class SurfaceFriction(StringConfigurable):
+    mjcf_configs: Optional[Dict[str, Any]] = None
+    sdf_configs: Optional[Dict[str, Any]] = None
+
+
+@dataclass
+class Surface(StringConfigurable):
+    contact: Optional[SurfaceContact] = None
+    friction: Optional[SurfaceFriction] = None
+
+
+@dataclass
+class Geom(StringConfigurable):
+    name: str
+    pose: Pose
+    shape: ShapeInfo
+    visual: Optional[VisualInfo] = None
+    surface: Optional[Surface] = None
+
+    mjcf_configs: Optional[Dict[str, Any]] = None
+    sdf_configs: Optional[Dict[str, Any]] = None
+
+    @property
+    def type(self):
+        return self.shape.type
+
+
+@dataclass
+class Joint(StringConfigurable):
+    name: str
+    parent: str
+    child: str
+    pose: Pose
+    joint_info: JointInfo
+    control_info: Optional[ControlInfo] = None
+
+    mjcf_configs: Optional[Dict[str, Any]] = None
+    sdf_configs: Optional[Dict[str, Any]] = None
+
+    @property
+    def type(self):
+        return self.joint_info.type
+
+    model: Optional["Model"] = None
+
+    def set_model(self, model):
+        self.model = model
+
+    @property
+    def parent_link(self):
+        return self.model.links[self.parent]
+
+    @property
+    def child_link(self):
+        return self.model.links[self.child]
```

Comment:
Removing for now.
Ideally, there should be a mapping from link name to Link.
But since the name for a link is enforced in SDL, this will trigger some issues.
I  will mark this as a to-do and implement it a bit later.